spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,990 @@
+require 'date'
+describe ManageIQ::Providers::Nuage::NetworkManager::VsdClient do
+  include Vmdb::Logging
+
+  UUID_REGEXP = /([0-9a-z]{8})-(([0-9a-z]{4})-){3}([0-9a-z]{12})/
+  ROUTE_PROPERTY_REGEXP = /([0-9]{5}):([0-9a-z]{4,5})/
+  IP_ADDR_REGEXP = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/
+  MAC_ADDR_REGEXP = /^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$/
+
+  ENTERPRISE_ID = '6abac3ad-a05d-4b93-9556-4ba5010fb13b'
+  DOMAIN_ID = 'd0c3274c-397a-4173-8981-bfd2f99ef8c6'
+  ZONE_ID = '76ac549a-5843-4f1b-ac90-b21ff6edc2a4'
+  SUBNET_ID = '4da803f7-c0d2-4beb-b02e-341ea77e377f'
+  POLICY_GROUP_ID = 'fadd09c4-9fea-46ec-8342-73f1b6a4df74'
+
+  before(:each) do
+    @userid = Rails.application.secrets.nuage_network.try(:[], 'userid') || 'NUAGE_USER_ID'
+    @password = Rails.application.secrets.nuage_network.try(:[], 'password') || 'NUAGE_PASSWORD'
+    @hostname = Rails.application.secrets.nuage_network.try(:[], 'host') || 'nuagenetworkhost'
+
+    # Ensure that VCR will obfuscate the basic auth
+    VCR.configure do |c|
+      # workaround for escaping host
+      c.before_playback do |interaction|
+        interaction.filter!(CGI.escape(@hostname), @hostname)
+        interaction.filter!(CGI.escape('NUAGE_NETWORK_HOST'), 'nuagenetworkhost')
+      end
+      c.filter_sensitive_data('NUAGE_NETWORK_AUTHORIZATION') { Base64.encode64("#{@userid}:#{@password}").chomp }
+    end
+  end
+
+  context "when login successful" do
+    before(:each) do
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/login', :record => :new_episodes) do
+        @vsd_client = described_class.new("https://#{@hostname}:8443/nuage/api/v5_0", @userid, @password)
+      end
+    end
+
+    it "should return valid non empty response for get_enterprises" do
+      enterprises = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/enterprises', :record => :new_episodes) do
+        enterprises = @vsd_client.get_enterprises
+      end
```

Comment:
Do you think we could wrap these four repeating lines into a function like:

```ruby
def vcr_get(cassette, method, args: [])
	res = nil
	VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/' + cassette) do
	  res = @vsd_client.send(method, *args)
	end
end
```

Then here we would replace them with:

```ruby
enterprises = vcr_get('enterprises', :get_enterprises)
```